core/Files.cc

```diff
@@ -113,7 +113,14 @@ File::Flags::Flags(string_view path)
 File::File(string &&path_, string &&source_, Type sourceType, uint32_t epoch)
     : epoch(epoch), sourceType(sourceType), flags(path_), packagedLevel{File::filePackagedSigil(source_)},
       path_(move(path_)), source_(move(source_)), originalSigil(fileStrictSigil(this->source_)),
-      strictLevel(originalSigil) {}
+      strictLevel(originalSigil) {
+    if (this->source_.size() > INT32_MAX) [[unlikely]] {
+        // Specifically looking for INT32_MAX, because File::lineBreaks_ stores `int`, not `uint32_t`
+        static_assert(sizeof(LocOffsets::beginLoc) == 4);
+        static_assert(is_same_v<remove_reference<decltype((*this->lineBreaks_)[0])>::type, int>);
+        Exception::raise("File larger than {} bytes. Got: {}", INT32_MAX, this->source_.size());
```

Comment:
I think the max size needs to be based off of `LocOffsets::INVALID_POS_LOC`?  It better be smaller than that, so we can't have valid files that have invalid offsets.

Reply:
I think that `INT32_MAX` is going to be smaller than `INVALID_POS_LOC`, which will effectively be `UINT32_MAX` because, despite `INVALID_POS_LOC` being `int`, it's stored into a `uint32_t` for `beginPos` and `endPos`, so the comparison will always happen at the unsigned version of the type.

The comment mentions that we want to check `INT32_MAX` because right now we store line break information in `vector<int>` instead of `vector<uint32_t>`, which means that line breaks for files larger than `2 ^ 31` bytes will get truncated (even before we get to the point where the size might collide with `INVALID_POS_LOC`).

Reply:
Looks like I'm mistaken, there are only 7 f's in the `INVALID_POS_LOC` definition, not 8, so `INVALID_POS_LOC` is actually `INT32_MAX`, meaning that if you have a file with exactly that many bytes in it, that's not okay? but if you have one or more bytes in your file, then it's back to being valid again 🤦

Reply:
wait it's even worse than that, `INT32_MAX` is `0x7fff'ffff`, so `INVALID_POS_LOC` is just somthing random: `2 ^ 28`